podigee/job/__main__.py

```diff
@@ -165,15 +170,66 @@ def get_podcast_metadata():
         "name": podcast_title
     }
 
+def get_end_date_on_granularity(granularity, start_date):
+    """
+    Get end date based on granularity and start date.
+    """
+    if granularity == "day":
+        return start_date
+    elif granularity == "month":
+        # last day of same month
+        return start_date + relativedelta(months=1) - timedelta(days=1)
+    return start_date
+
+def transform_podigee_podcast_overview(overview_data):
+    """
+    Transform Podigee podcast overview data to OpenPodcast format.
+    {"published_episodes_count":0,
+    "audio_published_minutes":0.0,
+    "unique_listeners_number":305,
+    "unique_subscribers_number":283,
+    "mean_audio_published_minutes":0,
+    "mean_episode_download":8.897435897435898,
+    "total_downloads":694.0,
+    "meta":{"from":"2025-08-01T00:00:00.000Z","to":"2025-08-31T23:59:59.999Z"}
+    """
+
+    metrics = [{
```

Comment:
This function's parsing makes me a little nervous because there's no error handling. Maybe we could introduce a type for it that we try to convert it into for added robustness?